plugins/modules/vmware_guest_instant_clone.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Ansible Project
+# Copyright: (c) 2021, Anant Chopra <chopraan@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: vmware_guest_instant_clone
+short_description: Instant Clone VM
+
+description:
+- This module can be used for Creating a powered-on Instant Clone of a virtual machine.
+- All variables and VMware object names are case sensitive.
+
+options:
+  name:
+    description:
+      - Name of the Cloned virtual machine.
+    type: str
+    aliases: ['vm_name']
+    required: True
+  uuid:
```

Comment:
Move UUID, moid and Use InstanceUUID paramters after parent_vm. User might get confused about vm_name and parent_vm